pkg/csi/service/wcpguest/controller.go

```diff
@@ -452,6 +479,78 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 	timeoutSeconds := int64(getAttacherTimeoutInMin(ctx) * 60)
 	timeout := time.Now().Add(time.Duration(timeoutSeconds) * time.Second)
 
+	// Check the feature state for file volume support
+	if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.FileVolume) {
+		// Feature is disabled on the cluster
+		return nil, status.Error(codes.InvalidArgument, "File volume not supported.")
+	}
+	// Retrieve Supervisor PVC
+	svPVC, err := c.supervisorClient.CoreV1().PersistentVolumeClaims(c.supervisorNamespace).Get(ctx, req.VolumeId, metav1.GetOptions{})
+	if err != nil {
+		msg := fmt.Sprintf("failed to retrieve supervisor PVC %q in %q namespace. Error: %+v", req.VolumeId, c.supervisorNamespace, err)
+		log.Error(msg)
+		return nil, status.Error(codes.Internal, msg)
+	}
+	for _, accessMode := range svPVC.Spec.AccessModes {
+		if accessMode == v1.ReadWriteMany || accessMode == v1.ReadOnlyMany {
+			// Build the CnsFileAccessConfig instance name and namespace
+			cnsFileAccessConfigInstance := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{}
+			cnsFileAccessConfigInstanceName := virtualMachine.Name + "-" + req.VolumeId
+			cnsFileAccessConfigInstanceKey := types.NamespacedName{
+				Namespace: c.supervisorNamespace,
+				Name:      cnsFileAccessConfigInstanceName,
+			}
+			if err := c.cnsOperatorClient.Get(ctx, cnsFileAccessConfigInstanceKey, cnsFileAccessConfigInstance); err != nil {
+				// Get() on the CnsFileAccessConfig instance failed with different error
+				msg := fmt.Sprintf("failed to get CnsFileAccessConfig instance: %q. Error: %+v", cnsFileAccessConfigInstance.Name, err)
+				log.Error(msg)
+				return nil, status.Errorf(codes.Internal, msg)
+			}
+			// Attacher timeout, default is set to 4 minutes
+			timeoutSeconds := int64(getAttacherTimeoutInMin(ctx) * 60)
+			// Adding watch on the CnsFileAccessConfig instance to register for updates
+			watchCnsFileAccessConfig, err := c.cnsFileAccessConfigWatcher.Watch(metav1.ListOptions{
```

Comment:
I'm not sure this filter works as expected.

Reply:
Seems to be working fine